File: 20. Valid Parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        vector<char> stack;
        map<char, char> d;
        d = {{'}', '{'},{']','['},{')', '('}};
        for(auto x : s){
            if(stack.size() == 0 || d[x] != stack.back()){
                stack.push_back(x);
            }
            else if(d[x] == stack.back()){
                stack.pop_back();
            }
        }
        return stack.size() == 0;
    }
};

File: 20. Valid Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        mapp = {")":"(", "}":"{", "]":"["}
        stack = []
        for ch in s:
            if ch not in mapp: stack.append(ch)
            elif stack and ch in mapp and stack[-1] == mapp[ch]: stack.pop()
            else: return False
        return len(stack) == 0
